lib/paper_trail/version_concern.rb

```diff
@@ -19,6 +19,20 @@ module VersionConcern
       belongs_to :item, polymorphic: true, optional: true
       validates_presence_of :event
       after_create :enforce_version_limit!
+
+      # As of Rails 7.0, when you call #versions on an item, and then call #item on one of the
+      # versions, you'll get back the exact same object as the original item. Two strange side
+      # effects result:
+      # 1. If the reifier uses version.item to reify, then it also modifies the original item
+      #    object. If you just deleted that item and it is frozen, then you'll get an error that
+      #    it can't modify a frozen object.
+      # 2. If you delete an item and then get the "last" version's item, you'll get the item you
+      #    just deleted rather than nil. That is because the belongs_to association's stale target
+      #    checking now thinks that the association hasn't gone stale and thus returns the
+      #    original item object.
+      if PaperTrail::RAILS_GTE_7_0
+        after_find { association(:item).instance_variable_set(:@stale_state, nil) }
```

Comment:
why not use the `inverse_of: nil` discussed in the other PR? it uses the public api of rails to fix it...

Reply:
Public api preferred of course. But doesn't work. At least, I just tried removing the `after_find` and adding `inverse_of: nil` to the `belongs_to :item`, and the specs failed that led me to this hack.

Reply:
Which may well indicate that this new Rails feature is broken, and until we can fix it...

Reply:
This `instance_variable_set` makes me pretty nervous. Pedro, if you're aware of a public-api alternative, can you please push a branch? If you can open a PR in David's fork, against this branch, that would be ideal. Thanks.

Reply:
`inverse_of: nil` didn't work but `inverse_of: false` did. Coming soon!

Reply:
And done.

Reply:
I guess now it's a question of how to get back those hours between setting `inverse_of` to `nil` (first thing I tried because it was the only obvious change from 6.1.4 to 7.0), and trying `false` instead.

Reply:
Nice!